ocs_ci/utility/metadata_utils.py

```diff
@@ -0,0 +1,258 @@
+    assert pod_obj.wait_for_resource(
+        condition=constants.STATUS_RUNNING,
+        selector="app=csi-cephfsplugin-provisioner",
+        dont_allow_other_resources=True,
+        timeout=60,
+    )
+
+    assert pod_obj.wait_for_resource(
+        condition=constants.STATUS_RUNNING,
+        selector="app=csi-rbdplugin-provisioner",
+        dont_allow_other_resources=True,
+        timeout=60,
+    )
+
+    # Check 'setmatadata' is set for csi-cephfsplugin-provisioner and csi-rbdplugin-provisioner pods
+    res = check_setmetadata_availability(pod_obj)
+    if not res:
+        raise AssertionError
+
+    cephfsplugin_provisioner_pods = pod.get_cephfsplugin_provisioner_pods()
+
+    args = pod_obj.exec_oc_cmd(
+        "get pod "
+        + cephfsplugin_provisioner_pods[0].name
+        + " --output jsonpath='{.spec.containers[4].args}'"
+    )
+    for arg in args:
+        if "--clustername" in arg:
+            return arg[14:]
+
+
+def disable_metadata(
+    config_map_obj,
+    pod_obj,
+):
+    """
+    Disable CSI_ENABLE_METADATA
+    Steps:
+    1:- Disable CSI_ENABLE_METADATA flag via patch request
+    2:- Check csi-cephfsplugin provisioner and csi-rbdplugin-provisioner
+    pods are up and running
+    3:- Check 'setmatadata' is not set for csi-cephfsplugin-provisioner
+    and csi-rbdplugin-provisioner pods
+
+    """
+    disable_metadata = '{"data":{"CSI_ENABLE_METADATA": "false"}}'
+
+    # Disable metadata feature for rook-ceph-operator-config using patch command
+    assert config_map_obj.patch(
+        resource_name="rook-ceph-operator-config",
+        params=disable_metadata,
+    ), "configmap/rook-ceph-operator-config not patched"
+
+    # Check csi-cephfsplugin provisioner and csi-rbdplugin-provisioner pods are up and running
+    assert pod_obj.wait_for_resource(
+        condition=constants.STATUS_RUNNING,
+        selector="app=csi-cephfsplugin",
+        dont_allow_other_resources=True,
+        timeout=60,
+    )
+
+    assert pod_obj.wait_for_resource(
+        condition=constants.STATUS_RUNNING,
+        selector="app=csi-rbdplugin",
+        dont_allow_other_resources=True,
+        timeout=60,
+    )
+
+    # Check 'setmatadata' is not set for csi-cephfsplugin-provisioner and csi-rbdplugin-provisioner pods
+    res = check_setmetadata_availability(pod_obj)
+    if res:
+        raise AssertionError
+
+
+def available_subvolumes(sc_name, toolbox_pod, fs):
+    """
+    To fetch available subvolumes for cephfs or rbd
+
+    Returns:
+    list: list of subvolumes available for rbd or cephfs
+
+    """
+    if sc_name == constants.DEFAULT_STORAGECLASS_CEPHFS:
+        # fs="ocs-storagecluster-cephfilesystem"
+        cephfs_subvolumes = toolbox_pod.exec_cmd_on_pod(
+            f"ceph fs subvolume ls {fs} --group_name csi"
+        )
+        log.info(f"available cephfs subvolumes-----{cephfs_subvolumes}")
+        return cephfs_subvolumes
+
+    else:
```

Comment:
what if sc_name is neither the default RBD SC or the default CephFS SC?
I suggest adding an explicit check (e.g. `elif sc_name == constants.DEFAULT_STORAGECLASS_RBD:`)
and an `else` with exception/return False